dateparser/languages/language.py

```diff
@@ -24,40 +26,51 @@ def __init__(self, shortname, language_info):
             if isinstance(value, int):
                 simplification[key] = str(value)
 
+        self._cached = self
```

Comment:
I wasn't sure if direct assignments to `self` like `self = SomeInstance()` a good practice or not. Hence, used `_cached` attrib. I tried implementing it on almost all levels. Higher levels slowed down the speed and had other issues but due to speed imperative I didn't look into fixing those issues. Tests took about 16 seconds to finish when I tried implementing it on DateDataParser level.